s64da_benchmark_toolkit/correctness.py

```diff
@@ -62,22 +76,17 @@ def check_for_mismatches(cls, truth, result):
         return mismatches
 
     def _check_correctness_impl(self, truth, result):
-        def prepare(df):
-            # Sort columns
-            df = df.sort_index(axis=1)
-            # Natsort all rows
-            df = df.reindex(index=order_by_index(df.index, index_natsorted(zip(df.to_numpy()))))
-            # Recreate index for comparison later
-            return df
 
         if truth.empty != result.empty:
             return list(result.index) if truth.empty else list(truth.index)
 
         if truth.shape != result.shape:
             return list(truth.index)
 
-        truth = prepare(truth)
-        result = prepare(result)
+        truth.drop_duplicates(inplace=True, ignore_index=True)
```

Comment:
If you're dropping duplicates, how do you know if you had the same amount of duplicates in both result sets?

Reply:
`if truth.shape != result.shape` checks that

Reply:
There are edge cases where this will fail: if, for example you have to duplicate rows in the truth set, and 2 different duplicate rows in the result set, it will report this as passing, even though they are not the same.
Granted this is a rare situation, question is, should we bother with this or not?

Reply:
You're right. But I think for now it's OK. If we're planning to introduce new, stable correctness checking in the future, we can leave it as it's.